platform/broadcom/sonic-platform-modules-dell/s5296f/sonic_platform/component.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
+
+########################################################################
+# DELLEMC S5296F
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the Components' (e.g., BIOS, CPLD, FPGA, BMC etc.) available in
+# the platform
+#
+########################################################################
+
+
+try:
+    import subprocess
+    from sonic_platform_base.component_base import ComponentBase
+    import sonic_platform.hwaccess as hwaccess
+
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+
+def get_bios_version():
+    return subprocess.check_output(['dmidecode', '-s',
+        'system-version']).decode('utf-8').strip()
+
+def get_fpga_version():
+    val = hwaccess.pci_get_value('/sys/bus/pci/devices/0000:04:00.0/resource0', 0)
+    return '{}.{}'.format((val >> 8) & 0xff, val & 0xff)
+
+def get_bmc_version():
+    """ Returns BMC Version """
+    bmc_ver = ''
+    try:
+        bmc_ver = subprocess.check_output(
+            "ipmitool mc info | awk '/Firmware Revision/ { print $NF }'",
+            shell=True, text=True).strip()
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        pass
+    return bmc_ver
+
+def get_cpld_version(bus, i2caddr):
+    """ Return CPLD version """
+    major = hwaccess.i2c_get(bus, i2caddr, 1)
+    minor = hwaccess.i2c_get(bus, i2caddr, 0)
+    if major != -1 and minor != -1:
+        return '{}.{}'.format(major, minor)
+    return ''
+
+def get_cpld0_version():
+    return get_cpld_version(601, 0x31)
+
+def get_cpld1_version():
+    return get_cpld_version(600, 0x30)
+
+def get_cpld2_version():
+    return get_cpld_version(600, 0x31)
```

Comment:
Are you sure get_cpld2_version( ) is same as get_cpld0_version()?

Reply:
Both are different, value is fetched from different registers.